homework/index.js

```diff
@@ -29,26 +14,170 @@
+    section.innerHTML = '';
+    const table = createAndAppend('table', section);
+    const firstRow = createTableRow(table, 'Repository:');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    createTableRow(table, 'Description:', {
+      text: repo.description ? repo.description : 'No description',
+    });
+    createTableRow(table, 'Forks:', { text: repo.forks });
+    createTableRow(table, 'Updated', { text: formatDate(repo.updated_at) });
+  }
+
+  function createListItem(ul, contributor) {
+    const li = createAndAppend('li', ul, { class: 'list-item' });
+    createAndAppend('img', li, {
+      src: contributor.avatar_url,
+      alt: contributor.login,
+      class: 'contributor-img',
+    });
+    createAndAppend('a', li, {
+      text: contributor.login,
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    createAndAppend('div', li, {
+      text: contributor.contributions,
+      class: 'contributions-div',
     });
   }
 
+  async function renderContribution(contributorsUrl, section) {
+    const contributors = await fetchJSON(contributorsUrl);
+    try {
+      section.innerHTML = '';
+      const ul = createAndAppend('ul', section);
+      const li = createAndAppend('li', ul);
+      createAndAppend('p', li, { text: 'Contributions' });
+      contributors.forEach(contributor => createListItem(ul, contributor));
+    } catch (err) {
+      createErrorMessage(err);
+    }
+    /* contributorsPromise
+      .then(data => {
+        section.innerHTML = '';
+        const ul = createAndAppend('ul', section);
+        const li = createAndAppend('li', ul);
+        createAndAppend('p', li, { text: 'Contributions' });
+        data.forEach(contributor => createListItem(ul, contributor));
+      })
+      .catch(err => {
+        createErrorMessage(err);
+      }); */
+  }
+
+  function selectRepository(select, mainDiv, repos) {
+    mainDiv.innerHTML = '';
+    const basicInfoSec = createAndAppend('section', mainDiv);
+    const contributionsSec = createAndAppend('section', mainDiv, {
+      class: 'contributions-section',
+    });
+    const repo = repos[select.value];
+
+    renderRepoDetails(repo, basicInfoSec);
+    renderContribution(repo.contributors_url, contributionsSec);
+  }
+
+  function renderRepoSelect(repo, optionValue, select) {
+    createAndAppend('option', select, {
+      text: repo.name,
+      value: optionValue,
+    });
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'main-header',
+    });
+    createAndAppend('div', header, {
+      class: 'header-text',
+      text: 'HYF-Repositories',
+    });
+    const select = createAndAppend('select', header, {
+      class: 'repo-select',
+    });
+    const mainSection = createAndAppend('main', root, { id: 'main' });
+
+    const repos = await fetchJSON(url);
```

Comment:
As per my previous comment, this function should be inside the `try` block as it can throw an error.